3_javascript/javascript1/week2/script.js

```diff
@@ -0,0 +1,96 @@
+// Flight booking fullname function
+
+function getFullname(firstname, surname, formalName) {
+  if (firstname.trim() === "" || surname.trim() === "") {
+    return "Please provide a valid firstname and surname.";
+  }
+
+  if (formalName) {
+    return formalName + " " + firstname + " " + surname;
+  } else {
+    return firstname + " " + surname;
+  }
+}
+
+const fullname1 = getFullname("Benjamin", "Hughes");
+const fullname2 = getFullname("Alice", "Smith", "Mrs");
+
+console.log(fullname1);
+console.log(fullname2);
+
+// Event application
+
+function getEventWeekday(daysFromToday) {
+  const daysOfWeek = [
+    "Monday",
+    "Tuesday",
+    "Wednesday",
+    "Thursday",
+    "Friday",
+    "Saturday",
+    "Sunday",
+  ];
+  const currentDate = new Date();
+  const currentDay = currentDate.getDay();
+
+  const eventDayIndex = (currentDay + daysFromToday) % 7;
+
+  const eventDayName = daysOfWeek[eventDayIndex];
+
+  return eventDayName;
+}
+
+// With todays weekday a Tuesday
+console.log(getEventWeekday(9));
+// With todays weekday a Friday
+console.log(getEventWeekday(2));
```

Comment:
```console.log(getEventWeekday(2));``` Should log out "Sunday" if the current day is Friday.  If I run the code when the current day is a Friday the result will be Monday.

This happens because ```currentDate.getDay();```returns the day starting on Sunday. So your daysOfWeek array should be modified to have Sunday as the first day of the array. It would look like this:

```
  const daysOfWeek = [
    "Sunday",
    "Monday",
    "Tuesday",
    "Wednesday",
    "Thursday",
    "Friday",
    "Saturday",
  ];
```